pkg/core/mempool/mem_pool.go

```diff
@@ -197,10 +218,63 @@ func (mp *Pool) Add(t *transaction.Transaction, fee Feer) error {
 	return nil
 }
 
+// checkAttributesConflicts is an internal unlocked method which returns an error in case
+// if transaction can not be added due to Conflicts attributes. It removes all conflicting
+// transactions from mempool in case if transaction has no conflicts or wins all conflicts.
+func (mp *Pool) checkAttributesConflicts(t *transaction.Transaction, fee Feer) error {
+	if fee.P2PSigExtensionsEnabled() {
+		var (
+			conflictingTxs []util.Uint256
+			ok             bool
+		)
+		// Step 1: check if `t` was in attributes of mempooled transactions
+		if conflictingTxs, ok = mp.conflicts[t.Hash()]; ok {
+			// check firstly if there are some mempooled transactions which have better network fee,
+			// if so, reject the new transaction
+			for _, hash := range conflictingTxs {
+				conflictingTx := mp.verifiedMap[hash]
+				if conflictingTx.HasSigner(t.Sender()) && conflictingTx.NetworkFee >= t.NetworkFee {
+					return fmt.Errorf("%w: conflicting transaction %s has bigger network fee", ErrConflictsAttribute, conflictingTx.Hash().StringBE())
+				}
+			}
+		}
+		// Step 2: check if mempooled transactions were in `t`'s attributes
+		for _, attr := range t.GetAttributes(transaction.ConflictsT) {
+			hash := attr.Value.(*transaction.Conflicts).Hash
+			existingTx, ok := mp.verifiedMap[hash]
+			if !ok {
+				continue
+			}
+			if !t.HasSigner(existingTx.Sender()) {
+				return fmt.Errorf("%w: not signed by the sender of conflicting transaction %s", ErrConflictsAttribute, existingTx.Hash().StringBE())
+			}
+			if existingTx.NetworkFee >= t.NetworkFee {
+				return fmt.Errorf("%w: conflicting transaction %s has bigger network fee", ErrConflictsAttribute, existingTx.Hash().StringBE())
+			}
+			// Otherwise add existingTx.Hash to the conflicts list in order to remove it from mempool at the end.
+			// Can't remove existingTx from mempool immediately because there still might be a transaction in the mempool
+			// which conflicts with the current tx and has bigger network fee.
+			conflictingTxs = append(conflictingTxs, existingTx.Hash())
+		}
+		// Step 3: new transaction was not rejected at the previous steps, so remove all mempooled
+		// conflicting transactions as they have smaller network fee
+		for _, hash := range conflictingTxs {
+			mp.removeInternal(hash, fee)
```

Comment:
This is a side-effect, even though it's called `checkAttributesConflicts`.

Reply:
I squashed `checkTxConflicts` and `checkAttributesConflicts`, because each time we use these two methods together. IMO, it looks better now.